hilti/runtime/include/types/struct.h

```diff
@@ -27,8 +27,23 @@ inline auto& value_or_exception(const std::optional<T>& t, const char* location)
 } // namespace struct_
 
 namespace detail::adl {
+
+template<typename>
+constexpr std::false_type has__str__helper(long);
+
+template<typename T>
+constexpr auto has__str__helper(int) -> decltype(std::declval<T>().__str__(), std::true_type{});
+
+template<typename T>
+using has__str__ = decltype(has__str__helper<T>(0));
+
 template<typename T, typename std::enable_if_t<std::is_base_of<trait::isStruct, T>::value>* = nullptr>
 inline std::string to_string(const T& x, adl::tag /*unused*/) {
```

Comment:
I am wondering if we add similar code to `to_string_for_print` which could forward to the the same hook (I don't think there's value in users customizing that one separately).

Reply:
`struct`` doesn't have a custom `to_string_for_print` currently, so seems we don't need that?

Reply:
Makes perfect sense.